table/block_based/block_based_table_builder.cc

```diff
@@ -1246,6 +1247,20 @@ void BlockBasedTableBuilder::CompressAndVerifyBlock(
         r->ioptions.clock,
         ShouldReportDetailedTime(r->ioptions.env, r->ioptions.stats));
 
+    *type = r->compression_type;
+#ifndef NDEBUG
+    if (r->compression_type != kNoCompression &&
+        g_hack_mixed_compression_in_block_based_table.LoadRelaxed() > 0U) {
+      // If zstd is in the mix, the compression_name table property needs to be
+      // set to it, for proper handling of context and dictionaries.
+      assert(!ZSTD_Supported() || r->compression_type == kZSTD);
```

Comment:
This part is not clear to me: when ZSTD is supported and the compression type is not ZSTD, we will fail the assertion so not to proceed? I thought we'd like to assert the opposite ... ?

Reply:
Ahh - I see 

> // If zstd is in the mix, the compression_name table property needs to be set to it, for proper handling of context and dictionaries.

This is not the reason to reject kZSTD now. It's just that we need to do something more in the future for kZSTD while still having the kZSTD in the mix.